### ISSUE TITLE
Anyone know how to do this .js bot?

### ISSUE THREAD

**Issue by buckman1974** [binary-com/binary-bot#922]
I don't know whether this forum is just for .xml bots, but wondered if anyone knew how to program a bot like this .js Firefox Macro bot?

It seems to martingale once, and auto change the LDP each time (I am not sure if there is any strategy to this or whether its just random). This user shows daily profits, which I am unsure are true, but looks like it could work?

I have contacted the owner but didn't get anywhere really.

I have a few .js bots but nothing the same as this, on auto? Any ideas?  :)

**Comment by nextnaro** [comment]
imacros

**Comment by buckman1974** [comment]
@nextnaro  yes.

**Comment by wilburforce83** [comment]
I don't know about that particular bot, but the next video that ran after piqued my interest ([here](https://www.youtube.com/watch?v=JJX9hMUQ6GM)). It seems effective but he is selling it for $15 a pop. It looks to me like he is just using @fruitfella 's last digit analyzer and a quite interesting money management strategy.

I don't believe in selling stuff that is freely available so I have made my own version, which I am almost certain is a close-as-dammit to identical as the one in the video, perhaps a slight variation on the money management, I notice his runs for longer than 5 recovery trades after losses within the recovery period, as well as increasing the stake. I have managed it slightly differently.

Please feel free to use and distribute for free :) 

[Download it here](https://gist.github.com/wilburforce83/c3129ac7086a9a80917cc1568d58d56e)

**Comment by buckman1974** [comment]
Thank you for this @wilburforce83   Good job!!  :)

Let me try it, but yes the seller was vague, and wanted $100+ for the .js bot, which I'm sure is floating around the web with other .js Binary bots, which maybe he changed a little..

he has numerous daily videos on Youtube showing this bot 'daily' making certain amounts, so was just interested in how he claims it won't ever go to a 2nd 'next' loss which i suppose is the main thing, as if you have the funds you'd recover the next martingale trade over any loss. (not sure how he does this, although he says the bot ''analyzes'' the market and digits at that current time) ?

**Comment by buckman1974** [comment]
Just trying this bot @wilburforce83 

Do you know where the Martingale is? as for digit differs, it looks like its set to x2 and differs, needs at least x10 unfortunately.

Secondly do you know which set of ''Last Digit Stats'' the ticks are running off? and how they are chosen? (is it the 0% ''Last Digit Stats'' and its chosen as an LDP?)
I just didn't know how the digits were chosen, as it just seemed to pick LDP 2, then LDP 4, for about 10 trades each?

**Comment by wilburforce83** [comment]
it is not a traditional martingale, it will run at x2 for 5 trades to recoup the loss, due to the high win rate it is the safest way to do it.

The last digits are chosen with @Fruifella 's last digit analyzer v2. You can set how many ticks to analyse in the config of the bot. Let me know you are stuck and I can help you out a bit more, just getting my daughter to bed atm!

**Comment by buckman1974** [comment]
Yes please, @wilburforce83 ... I am just testing the bot now and now understand the x2 for 5 trades to recoup the loss, which maybe better than one trade to martingale, to cover a loss?
Although you just have to make sure one of your 5 x recover trades doesn't lose!  :)

Any help on the @Fruifella 's last digit analyzer v2 would be good.. It looks complicated, but tried to change the analyzed ticks to 9, and then 20 (so they move around a little) but its now only choosing a differ with LDP 1 only? (so not sure what i did lol) so obviously with it just picking '1' it started to lose..

(I'm not sure how to PM or direct message on here so forgive me for posting)  :)

**Comment by davidtran89** [comment]
> I don't know about that particular bot, but the next video that ran after piqued my interest ([here](https://www.youtube.com/watch?v=JJX9hMUQ6GM)). It seems effective but he is selling it for $15 a pop. It looks to me like he is just using @Fruitfella 's last digit analyzer and a quite interesting money management strategy.
> 
> I don't believe in selling stuff that is freely available so I have made my own version, which I am almost certain is a close-as-dammit to identical as the one in the video, perhaps a slight variation on the money management, I notice his runs for longer than 5 recovery trades after losses within the recovery period, as well as increasing the stake. I have managed it slightly differently.
> 
> Please feel free to use and distribute for free :)
> 
> [Download it here](https://gist.github.com/wilburforce83/c3129ac7086a9a80917cc1568d58d56e)
